cma/evolution_strategy.py

```diff
@@ -4646,4 +4664,36 @@ def update(es):
     es.objective_function_complements = [_al]
     es.augmented_lagrangian = _al
     es.best_feasible = best_feasible_solution
+
+    if post_optimization:
+        if h != no_constraints and (not isinstance(post_optimization, float) or post_optimization <= 0):
+            raise ValueError("In in case when equality constraints are present, "
+                             "then post_optimization must be a strictly positive "
+                             "float indicating the error on the inequality constraints")
+
+        kwargs_post_opt = kwargs.copy()
+        if 'options' in kwargs_post_opt:
+            kwargs_post_opt['options']['ftarget'] = 0
+        else:
+            kwargs_post_opt['options'] = {'ftarget': 0}
+
+        _, es_post_opt = fmin2(lambda x: sum([gi ** 2 if gi > 0 else 0 for gi in g(x)]) +
+                                         sum([hi ** 2 if hi ** 2 > post_optimization ** 2 else 0 for hi in h(x)]),
+                               es.result.xfavorite, sigma0 / 1000, **kwargs_post_opt)
+        x_post_opt = es_post_opt.result.xfavorite
+        g_x_post_opt, h_x_post_opt = g(x_post_opt), h(x_post_opt)
+        if all([gi <= 0 for gi in g_x_post_opt]) and \
+                all([hi ** 2 <= post_optimization ** 2 for hi in h_x_post_opt]):
+            f_x_post_opt = objective_function(x_post_opt)
+            es.best_feasible_post_opt = ot.BestSolution2()
```

Comment:
Why not simply using `es.best_feasible_solution.update(...)`? The user generally wouldn't want to search for the best solution under different attributes.

Reply:
Yes, I understand I didn't update the `best_feasible` attribute at first because I thought that the user might want to keep the two feasible solutions (the one from the first optimization step and the one from the post-optimization step).
But actually, the user can access them both thanks to the `previous` field in `es.best_feasible` so I used `es.best_feasible.update(...)` in the last commit.